### PR TITLE
Allows the app to run if screen is off

### PR DESCRIPTION
**Pull request by kurly1**
Added the option to let WhereYouGo run in background if the screen of the phone is off.

For this change only english and german translations are includet.

Please review this code change

### CODE REVIEW

File: src/main/res/values/strings_pref.xml [hunk 1]
```diff
@@ -8,6 +8,8 @@
     <string name="pref_confirm_on_exit_desc">Display confirmation dialog before exiting Locus via the Back Button</string>
     <string name="pref_double_click">Double tap</string>
     <string name="pref_double_click_desc">To exit the game use double tap on BACK</string>
+    <string name="pref_run_screen_off">Run app if screen is off</string>
+    <string name="pref_run_screen_off_desc">Ensures that the app is running even if the screenis off.</string>
```
Comment (bekuno) [inline src/main/res/values/strings_pref.xml]: typo: screenis -> screen is

File: src/main/java/menion/android/whereyougo/geo/location/LocationState.java [hunk 2]
```diff
@@ -13,6 +13,10 @@
  * see <http://www.gnu.org/licenses/>.
  * 
  * Copyright (C) 2012 Menion <whereyougo@asamm.cz>
+ *
+ * Changes:
+ * Date        Who               Detail
+ * 19.01.2020  Kurly1            Add support for partial WakeLock
```
Comment (bekuno) [inline src/main/java/menion/android/whereyougo/geo/location/LocationState.java]: We use the git history to find out this infos. 
Please remove.

File: src/main/java/menion/android/whereyougo/preferences/PreferenceValues.java [hunk 3]
```diff
@@ -13,6 +13,11 @@
  * see <http://www.gnu.org/licenses/>.
  * 
  * Copyright (C) 2012 Menion <whereyougo@asamm.cz>
+ *
+ * Changes:
+ * Date        Who        Detail
+ * 19.01.2018  Kurly1     Replaced enableWakeLock function tu support partial WakeLock
+ *                        to allow the application to run if screen is off.
```
Comment (bekuno) [inline src/main/java/menion/android/whereyougo/preferences/PreferenceValues.java]: We use the git history to find out this infos. 
Please remove.

File: src/main/java/menion/android/whereyougo/preferences/Preferences.java [hunk 4]
```diff
@@ -13,6 +13,10 @@
  * see <http://www.gnu.org/licenses/>.
  * 
  * Copyright (C) 2012 Menion <whereyougo@asamm.cz>
+ *
+ * Changes:
+ * Date         Who                Detail
+ * 19.01.2020   Kurly1             Add support for partial WakeLock (KEY_B_RUN_SCREEN_OFF)
```
Comment (bekuno) [inline src/main/java/menion/android/whereyougo/preferences/Preferences.java]: We use the git history to find out this infos. 
Please remove.

File: src/main/res/values-de/strings_pref.xml [hunk 5]
```diff
@@ -8,6 +8,8 @@
     <string name="pref_confirm_on_exit_desc">Zeigt einen Dialog bevor Locus mit dem Zurück-Knopf beendet wird</string>
     <string name="pref_double_click">Doppelklicken</string>
     <string name="pref_double_click_desc">ZURÜCK doppelklicken um das Spiel zu beenden</string>
+    <string name="pref_run_screen_off">App läuft bei ausgeschaltetem Display</string>
+    <string name="pref_run_screen_off_desc">Stellt sicher, dass die App auch wenn das Display ausgeschaltet ist läuft.</string>
```
Comment (bekuno) [inline src/main/res/values-de/strings_pref.xml]: Better grammer:
Stellt sicher, dass die App auch dann läuft, wenn das Display ausgeschaltet ist.

File: src/main/java/menion/android/whereyougo/geo/location/LocationState.java (hunk 2 above)
Comment (kurly1) [reply src/main/java/menion/android/whereyougo/geo/location/LocationState.java]: I removed the change log from the files.

File: src/main/java/menion/android/whereyougo/preferences/PreferenceValues.java (hunk 3 above)
Comment (kurly1) [reply src/main/java/menion/android/whereyougo/preferences/PreferenceValues.java]: I removed the change log from the files.

File: src/main/java/menion/android/whereyougo/preferences/Preferences.java (hunk 4 above)
Comment (kurly1) [reply src/main/java/menion/android/whereyougo/preferences/Preferences.java]: I removed the change log from the files.

File: src/main/res/values/strings_pref.xml (hunk 1 above)
Comment (kurly1) [reply src/main/res/values/strings_pref.xml]: Typo and grammer have been fixed.

File: src/main/res/values-de/strings_pref.xml (hunk 5 above)
Comment (kurly1) [reply src/main/res/values-de/strings_pref.xml]: Typo and grammer have been fixed.

**Comment by kurly1** [comment]
I testet the code on API25 (Android 7.1.1) and API27 (Android 8.1).

**Comment by bekuno** [comment]
@Lineflyer I can't test this PR on a android 9 or 10. 
That's why I think that we should create a new version of WhereYouGo only with PR #2 and #37 36. Afterwards we can develop a maintenance version with this PR.

**Comment by bekuno** [comment]
OK to test

**Comment by kumy** [comment]
retest this please

**Comment by Lineflyer** [comment]
Please do not merge this yet, until we merged some bug fixes (which are currently targeted to the master branch) and merged to `release` once again.

**Comment by Lineflyer** [comment]
Open bugfix PRs have now been retargeted to `release`. So whenever ready this PR can be merged to `master` as we do not need to merge `master` to `release` for the needed bugfixes anymore. Removing label.

**Comment by bekuno** [comment]
Lets give it a try in the nightlies.

**Pull request merged**